docs/design/events.md

```diff
@@ -0,0 +1,60 @@
+# Events
+
+An Event is the entire interaction of the UOR Client with a referenced Collection. Since a Collection can have many nested Collections, a single event is the complete interaction with all Collections referenced by the UOR Client.
+
+Collections have embedded Event Engines. An Event Engine contains the application logic of a Collection. 
+
+## Event Engine Lookup
+
+1. UOR Client retrieves Collection root node
+2. UOR Client looks up Schema in Collection root node
+3. UOR Client retrieves the Schema Collection root node
+4. UOR Client looks up application logic in the Schema Collection root node
+5. UOR Client retrieves the application logic collection root node
+6. UOR Client looks up application logic by platform/arch
+7. UOR Client retrieves the application logic by platform/arch
+
+
+## Router
+
+When the UOR Client pushes or pulls (an object), it performs a routing action. The UOR Client can perform 2 routing actions:
+
+1. Route object to Collection
+2. Route object to Event Engine
+
+
+### Router API
+| Control Message   |
+|-------------------|
+| attributes (map)  | The attributes of the object
+| digest            | The digest of the object
+| source            | The source address of the object
+| destination       | The destination address of the object
+| payload (bool)    | If no payload, the object is only attributes
+
+# Event Engine Design
+
+Event Engines are designed as follows:
+1. Event Engines can be UOR native applications or legacy applications ported to UOR.
+2. An application becomes an Event Engine when it implements the UOR Router API and performs tasks pertinent to a UOR Collection. 
```

Comment:
Is the Router an Interface? I see a Control Message, but what else is there defined aside from the 2 routing actions and the Control Message

Also, what are tasks "pertinent" to a UOR collection?